nmt.go

```diff
@@ -142,35 +181,59 @@ func (n *NamespacedMerkleTree) ProveRange(start, end int) (Proof, error) {
 
 // ProveNamespace returns a range proof for the given NamespaceID.
 //
-// In case the underlying tree contains leaves with the given namespace
-// their start and end index will be returned together with a range proof and
-// the found leaves. In that case the returned leafHash will be nil.
+// case 1) If the namespace nID is out of the range of the tree's min and max namespace
+// i.e., (nID < n.minNID) or (n.maxNID < nID)
+// ProveNamespace returns an error and does not generate any range proof,
```

Comment:
[blocking][question] This claims to return an error but the function doesn't return an error

```golang
	if nID.Less(n.minNID) || n.maxNID.Less(nID) {
		return NewEmptyRangeProof(isMaxNsIgnored), nil
	}
```

Should the doc or the implementation be updated?

Reply:
Good catch, will investigate it in the following tracking issue https://github.com/celestiaorg/nmt/issues/99.